<launch>


<arg name="bagfile" default="/home/xesc/BAGFILES_BAG/cabrera_16092020/16092020_163754/images_2020-09-16-16-37-55_0.bag
      /home/xesc/BAGFILES_BAG/cabrera_16092020/16092020_163754/topics_2020-09-16-16-37-56_0.bag"/> 


  <param name="/use_sim_time" value="true"/>
  <arg name="camera1" default="/stereo_down"/>

  <node pkg="rosbag" type="play" name="rosbag" args="--clock -r 0.8 $(arg bagfile)" />
  <arg name="disparity_params" value="$(find cola2_turbot)/config/avt_manta/disparity_params_x1.yaml" />


  <!-- image proc: debayer and rectification  --> 

  <node pkg="stereo_image_proc" type="stereo_image_proc" name="image_proc_left" respawn="true" output="screen">
    <remap from="left/image_raw" to="$(arg camera1)/left/image_raw"/>
    <remap from="left/camera_info" to="$(arg camera1)/left/camera_info"/>
    <remap from="left/image_rect_color" to="$(arg camera1)/left/image_rect_color"/>
    <remap from="right/image_raw" to="$(arg camera1)/right/image_raw"/>
    <remap from="right/camera_info" to="$(arg camera1)/right/camera_info"/>
    <remap from="right/image_rect_color" to="$(arg camera1)/right/image_rect_color"/>
    <remap from="disparity" to="$(arg camera1)/disparity"/>
    <remap from="points2" to="$(arg camera1)/points2"/>
    <rosparam file="$(arg disparity_params)"/>
    <param name="approximate_sync" value="true" />
  </node>



<node name="rviz" pkg="rviz" type="rviz"/>

</launch>
